app/backend/src/couchers/couchers_select.py

```diff
@@ -38,20 +54,43 @@ def filter_by_username_or_id(self, field):
         # no fields match, this will return no rows
         return self.filter(False)
 
-    def filter_users(self, context, table=User):
+    def filter_users(self, session, context, table=User):
+        """
+        Filters out users that should not be visible: blocked, deleted, or banned
+
+        Filters the given table, assuming it's already joined/selected from
+        """
+        hidden_users = _blocked_users(session, context.user_id)
+        return self.filter(table.is_visible).filter(~table.id.in_(hidden_users))
```

Comment:
It would be fantastic if we could use subqueries for this to make the select query construction independent of the actual session.

Reply:
Makes way more sense. Done.